google/compute_beta_operation.go

```diff
@@ -0,0 +1,167 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+
+	computeBeta "google.golang.org/api/compute/v0.beta"
+)
+
+// OperationBetaWaitType is an enum specifying what type of operation
+// we're waiting on from the beta API.
+type ComputeBetaOperationWaitType byte
+
+const (
+	ComputeBetaOperationWaitInvalid ComputeBetaOperationWaitType = iota
+	ComputeBetaOperationWaitGlobal
+	ComputeBetaOperationWaitRegion
+	ComputeBetaOperationWaitZone
+)
+
+type ComputeBetaOperationWaiter struct {
+	Service *computeBeta.Service
+	Op      *computeBeta.Operation
+	Project string
+	Region  string
+	Type    ComputeBetaOperationWaitType
+	Zone    string
+}
+
+func (w *ComputeBetaOperationWaiter) RefreshFunc() resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		var op *computeBeta.Operation
+		var err error
+
+		switch w.Type {
+		case ComputeBetaOperationWaitGlobal:
+			op, err = w.Service.GlobalOperations.Get(
+				w.Project, w.Op.Name).Do()
+		case ComputeBetaOperationWaitRegion:
+			op, err = w.Service.RegionOperations.Get(
+				w.Project, w.Region, w.Op.Name).Do()
+		case ComputeBetaOperationWaitZone:
+			op, err = w.Service.ZoneOperations.Get(
+				w.Project, w.Zone, w.Op.Name).Do()
+		default:
+			return nil, "bad-type", fmt.Errorf(
+				"Invalid wait type: %#v", w.Type)
+		}
+
+		if err != nil {
+			return nil, "", err
+		}
+
+		log.Printf("[DEBUG] Got %q when asking for operation %q", op.Status, w.Op.Name)
+
+		return op, op.Status, nil
+	}
+}
+
+func (w *ComputeBetaOperationWaiter) Conf() *resource.StateChangeConf {
+	return &resource.StateChangeConf{
+		Pending: []string{"PENDING", "RUNNING"},
+		Target:  []string{"DONE"},
+		Refresh: w.RefreshFunc(),
+	}
+}
+
+// ComputeBetaOperationError wraps computeBeta.OperationError and implements the
+// error interface so it can be returned.
+type ComputeBetaOperationError computeBeta.OperationError
+
+func (e ComputeBetaOperationError) Error() string {
+	var buf bytes.Buffer
+
+	for _, err := range e.Errors {
```

Comment:
See above. If it's something you think would be valuable for `compute_operation.go` as well, feel free to comment with it on #191.